### PR TITLE
Add --stdin-filename option to allow naming the text being processed on STDIN

### LINKED ISSUE #1950: Allow one filename when using --stdin
Editor-integrations can use a combination of `--stdin` and the current working directory to lint a file that has yet to be saved - but this means that any rule which relies on the filename doesn't work.

See https://github.com/roadhump/SublimeLinter-eslint/issues/28 for some discussion on this.

I'm proposing that:
* Passing multiple filenames alongside `--stdin` will raise an error
* Passing a single filename alongside `--stdin` will use that filename for linting

Currently filenames passed at the same time as `--stdin` are ignored.

Happy to submit a PR for this.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/9268027-allow-one-filename-when-using-stdin?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by tikotzky**
This PR should fix #1950
@nzakas please let me know if you have any feedback or would like me to make any changes.

### CODE REVIEW

**Comment by ilyavolodin** [comment]
Could you change the message of your commit to follow our guidelines listed here: http://eslint.org/docs/developer-guide/contributing#step-2:-make-your-changes and squash them?

**Comment by tikotzky** [comment]
done. sorry about that.

File: lib/cli.js [hunk 1]
```diff
@@ -175,7 +175,7 @@ var cli = {
             engine = new CLIEngine(translateOptions(currentOptions));
             debug("Running on " + (text ? "text" : "files"));
 
-            result = text ? engine.executeOnText(text) : engine.executeOnFiles(files);
+            result = text ? engine.executeOnText(text, currentOptions.stdinFilename) : engine.executeOnFiles(files);
             if (currentOptions.quiet) {
```
Comment (nzakas) [inline lib/cli.js]: If this option isn't passed, it should default to "<text>" as it was before.

Comment (tikotzky) [inline lib/cli.js]: You would like `currentOptions.stdinFilename` to be defaulted to `""`?
The way it is now if `--stdin-filename` isnt passed then `currentOptions.stdinFilename` will be undefined. 
`filename` will then get defaulted to `"<text>"` [here](https://github.com/tikotzky/eslint/blob/add-stdin-filename-option/lib/cli-engine.js#L225)

Would you like me to take a different approach?

Comment (nzakas) [inline lib/cli.js]: Ah, I completely missed that. This approach looks fine.

**Comment by nzakas** [comment]
This looks good. Can you please sign our CLA? http://eslint.org/cla

**Comment by tikotzky** [comment]
signed :smile:

**Pull request merged**